tests/subscriptions/test_filter_subscriptions.py

```diff
@@ -60,3 +77,74 @@ def test_filter_subscriptions(expected_subs, extra_subs) -> None:  # type: ignor
         slice_id=2,
     )
     assert filtered_subs == expected_subs
+
+
+def build_rpc_subscription(resolution: timedelta, org_id: int) -> Subscription:
+    return Subscription(
+        SubscriptionIdentifier(PartitionId(1), uuid.uuid4()),
+        RPCSubscriptionData.from_proto(
+            CreateSubscriptionRequestProto(
+                time_series_request=TimeSeriesRequest(
+                    meta=RequestMeta(
+                        project_ids=[1],
+                        organization_id=org_id,
+                        cogs_category="something",
+                        referrer="something",
+                    ),
+                    aggregations=[
+                        AttributeAggregation(
+                            aggregate=Function.FUNCTION_SUM,
+                            key=AttributeKey(
+                                type=AttributeKey.TYPE_FLOAT, name="test_metric"
+                            ),
+                            label="sum",
+                            extrapolation_mode=ExtrapolationMode.EXTRAPOLATION_MODE_SAMPLE_WEIGHTED,
+                        ),
+                    ],
+                    filter=TraceItemFilter(
+                        comparison_filter=ComparisonFilter(
+                            key=AttributeKey(type=AttributeKey.TYPE_STRING, name="foo"),
+                            op=ComparisonFilter.OP_NOT_EQUALS,
+                            value=AttributeValue(val_str="bar"),
+                        )
+                    ),
+                ),
+                time_window_secs=300,
+                resolution_secs=int(resolution.total_seconds()),
+            ),
+            EntityKey.EAP_SPANS,
+        ),
+    )
+
+
+@pytest.fixture
+def expected_rpc_subs() -> MutableSequence[Subscription]:
+    return [
+        build_rpc_subscription(timedelta(minutes=1), 2)
+        for count in range(randint(1, 50))
+    ]
+
+
+@pytest.fixture
+def extra_rpc_subs() -> MutableSequence[Subscription]:
+    return [
+        build_rpc_subscription(timedelta(minutes=3), 1)
+        for count in range(randint(1, 50))
+    ]
+
+
+@patch("snuba.settings.SLICED_STORAGE_SETS", {"events_analytics_platform": 3})
+@patch(
+    "snuba.settings.LOGICAL_PARTITION_MAPPING",
+    {"events_analytics_platform": {0: 0, 1: 1, 2: 2}},
+)
+def test_filter_rpc_subscriptions(expected_rpc_subs, extra_rpc_subs) -> None:  # type: ignore
+    importlib.reload(scheduler)
+
+    filtered_subs = filter_subscriptions(
+        subscriptions=expected_rpc_subs + extra_rpc_subs,
+        entity_key=EntityKey.EAP_SPANS,
+        metrics=DummyMetricsBackend(strict=True),
+        slice_id=2,
+    )
+    assert filtered_subs == expected_rpc_subs
```

Comment:
this test is not necessary.

1. eap_spans is not a sliced storage
2. this is far in the weeds of the implementation details of the subscription scheduler